Makefile

```diff
@@ -0,0 +1,41 @@
+MSBUILD = xbuild
+CONFIGURATION = Release
+BIN = bin/$(CONFIGURATION)
+MDOC = $(BIN)/mdoc.exe
+
+all: build
+
+build: $(MDOC)
+
+$(MDOC): $(BIN)/ICSharpCode.SharpZipLib.dll $(BIN)/Mono.Cecil.dll
+	$(MSBUILD) apidoctools.sln /p:Configuration=$(CONFIGURATION)
+	cp mdoc/$(BIN)/mdoc.* $(BIN)/
```

Comment:
I think the `.csproj` should build directly into `bin/$(Configuration)`, not have a separate build step.

For the `external/*` directories, this isn't really an option -- you don't want to have to modify them -- but for `.csproj` files in *this* repo, it's straightforward to have them do:

    <OutputPath>..\bin\Debug</OutputPath>

This would avoid the need for an "extra" `cp` in the Makefile, and help keep things consistent.

*Similar* logic *can* apply to the contents of `external`, in that you might be able to override the `$(OutputPath)` property:

    $(MSBUILD) external/SharpZipLib/ICSharpCode.SharpZipLib.NET45/ICSharpCode.SharpZipLib.csproj \
        /p:OutputPath=..\..\..\bin\$(CONFIGURATION) \
        /p:Configuration=Release

However, in my experience that *occasionally* goes "wrong," so you'd need to test+verify first.